Source/WebCore/Modules/mediasession/MediaSession.cpp

```diff
@@ -245,8 +245,14 @@ void MediaSession::setPlaybackState(MediaSessionPlaybackState state)
     notifyPlaybackStateObservers();
 }
 
-void MediaSession::setActionHandler(MediaSessionAction action, RefPtr<MediaSessionActionHandler>&& handler)
+ExceptionOr<void> MediaSession::setActionHandler(MediaSessionAction action, RefPtr<MediaSessionActionHandler>&& handler)
 {
+#if ENABLE(MEDIA_STREAM)
+    RefPtr document = this->document();
+    if (document && !document->settings().mediaSessionCaptureToggleAPIEnabled() && (action == MediaSessionAction::Togglecamera || action == MediaSessionAction::Togglemicrophone || action == MediaSessionAction::Togglescreenshare))
```

Comment:
shouldn't we throw if document is null ?
so`if ((!document || !document->settings().mediaSessionCaptureToggleAPIEnabled()) && ...`

Reply:
It probably should but this should be done in a follow-up patch, not this one.

Reply:
Chrome is not throwing either, so it seems best to keep the same behaviour in this patch.